### PR TITLE
Added explicit setting of the SDK version to ILogger trace calls.

### PR DESCRIPTION
**Pull request by dnduffy**

### CODE REVIEW

**Comment by dnduffy** [comment]
I don't think this actually fixes the problem I was trying to address because I still get telemetry with the base SDK version being reported which doesn't go through this path.  If I recall correctly this logger isn't used except in debug mode with no iKey.

File: src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLogger.cs [hunk 1]
```diff
@@ -41,19 +45,21 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
...
                 {
-                    foreach (var item in stateDictionary)
+                    foreach (KeyValuePair<string, object> item in stateDictionary)
                     {
                         dict[item.Key] = Convert.ToString(item.Value);
                     }
                 }
 
-                this.telemetryClient.TrackTrace(formatter(state, exception), this.GetSeverityLevel(logLevel), dict);
+                traceTelemetry.Context.GetInternalContext().SdkVersion = this.sdkVersion;
```
Comment (pakrym) [inline src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLogger.cs]: Why not have initializer doing this instead?

**Comment by pakrym** [comment]
Tests?

File: src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLogger.cs (hunk 1 above)
Comment (SergeyKanzhelev) [inline src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLogger.cs]: Initializers will set the version for all telemetry item - custom reported, collected by dependency collector module, etc. We want to distinguish those so we can help troubleshoot the issue when we see "bad" trace telemetry. Knowing how that "bad" trace was collected is essential

Comment (pakrym) [inline src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLogger.cs]: Then I would prefer having this `traceTelemetry.Context.GetInternalContext().SdkVersion = ` and ` this.sdkVersion = SdkVersionUtils.VersionPrefix + SdkVersionUtils.GetAssemblyVersion();` in common utility class. Just don't like increasing code duplication

Comment (dnduffy) [inline src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLogger.cs]: I'll create an issue to track that cleanup as an enhancement for the next build.

**Comment by dnduffy** [comment]
Unit test added.

**Pull request merged**